src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -14,211 +24,3020 @@ namespace Microsoft.CodeAnalysis.CSharp
+        /// </summary>
+        protected readonly BoundNode methodMainNode;
+
+        /// <summary>
+        /// The flow analysis state at each label, computed by merging the state from branches to
+        /// that label with the state when we fall into the label.  Entries are created when the
+        /// label is encountered.  One case deserves special attention: when the destination of the
+        /// branch is a label earlier in the code, it is possible (though rarely occurs in practice)
+        /// that we are changing the state at a label that we've already analyzed. In that case we
+        /// run another pass of the analysis to allow those changes to propagate. This repeats until
+        /// no further changes to the state of these labels occurs.  This can result in quadratic
+        /// performance in unlikely but possible code such as this: "int x; if (cond) goto l1; x =
+        /// 3; l5: print x; l4: goto l5; l3: goto l4; l2: goto l3; l1: goto l2;"
+        /// </summary>
+        private readonly PooledDictionary<LabelSymbol, TLocalState> _labels;
+
+        /// <summary>
+        /// Set to true after an analysis scan if the analysis was incomplete due to state changing
+        /// after it was used by another analysis component.  In this case the caller scans again (until
+        /// this is false). Since the analysis proceeds by monotonically changing the state computed
+        /// at each label, this must terminate.
+        /// </summary>
+        protected bool stateChangedAfterUse;
+
+        /// <summary>
+        /// All of the labels seen so far in this forward scan of the body
+        /// </summary>
+        private PooledHashSet<BoundStatement> _labelsSeen;
+
+        /// <summary>
+        /// If we are tracking exceptions, then by convention the first entry in the pending branches
+        /// buffer contains a summary of the states that can arise from exceptions.
+        /// </summary>
+        private readonly bool _trackExceptions;
+
+        /// <summary>
+        /// Pending escapes generated in the current scope (or more deeply nested scopes). When jump
+        /// statements (goto, break, continue, return) are processed, they are placed in the
+        /// pendingBranches buffer to be processed later by the code handling the destination
+        /// statement. As a special case, the processing of try-finally statements might modify the
+        /// contents of the pendingBranches buffer to take into account the behavior of
+        /// "intervening" finally clauses.
+        /// </summary>
+        protected ArrayBuilder<PendingBranch> PendingBranches { get; private set; }
+
+        /// <summary>
+        /// The definite assignment and/or reachability state at the point currently being analyzed.
+        /// </summary>
+        protected TLocalState State;
+        protected TLocalState StateWhenTrue;
+        protected TLocalState StateWhenFalse;
+        protected bool IsConditionalState;
```

Comment:
what does IsConditionalState mean?

Reply:
That's when we're in the middle of a split state.